### PR TITLE
Add Item count mode 

### PR DESCRIPTION
**Pull request by gustavopvasconcellos**
#### What problem is this solving?

The unmatching of the items quantity logic on minicart

image

#### How to test it?

[Workspace](https://ecom6853--carrefourbrfood.myvtex.com/)

#### Screenshots or example usage:

image

#### Related to / Depends on

It related to: https://github.com/vtex-apps/minicart/pull/249

### CODE REVIEW

File: react/ProductList.tsx [hunk 1]
```diff
@@ -111,8 +112,37 @@ const ProductGroup: StorefrontFunctionComponent<Props> = props => {
...
+      return itemQuantity
+    }, 0),
+    total: arr.reduce((itemQuantity: number, item: Item) => {
+      return itemQuantity + item.quantity
+    }, 0),
+    distinct: arr.length,
+  }
+
+  return options[countMode] ?? options.distinct
+}
```
Comment (kaisermann) [inline react/ProductList.tsx]: While this function works as expected, it unnecessarily loops the `arr` multiple times for any count mode. Can you refactor to multiple `if(countMode === '...') { return arr.reduce(...)} `?

File: react/ProductList.tsx [hunk 2]
```diff
@@ -111,8 +112,37 @@ const ProductGroup: StorefrontFunctionComponent<Props> = props => {
   )
 }
 
+type TotalItemsType =
+  | 'total'
+  | 'distinct'
+  | 'totalAvailable'
+  | 'distinctAvailable'
+
```
Comment (kaisermann) [inline react/ProductList.tsx]: Nitpick: this should be put next to the other types of this file.

File: react/ProductList.tsx [hunk 3]
```diff
@@ -111,8 +112,37 @@ const ProductGroup: StorefrontFunctionComponent<Props> = props => {
...
+const countModeHandle = (countMode: TotalItemsType, arr: Item[]) => {
+  const options = {
+    distinctAvailable: arr.reduce((itemQuantity: number, item: Item) => {
+      if (item.availability === 'available') {
+        return ++itemQuantity
```
Comment (kaisermann) [inline react/ProductList.tsx]:
```suggestion
        return itemQuantity + 1
```

**Review by kaisermann — CHANGES_REQUESTED** [review]

**Comment by gustavopvasconcellos** [comment]
@kaisermann I've done all the changes, thank you

**Review by kaisermann — APPROVED** [review]

File: docs/README.md [hunk 4]
```diff
@@ -264,6 +264,12 @@ Therefore, in order to customize the `product-list` configuration, you can simpl
...
 
+### `product-list` props
+
+| Prop name              | Type      | Description                                                                                                                                                                                                                                     | Default value |
+| ---------------------- | --------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
+| `itemCountMode` | `enum` | Quantity badge behavior when displaying the number of total items added in Minicart. Possible values are: `total`  (quantity badge displays the number of items added to the cart), `distinct` (quantity badge only displays the number of different products added to the cart), `totalAvailable`  (quantity badge displays the number of available items added to the cart) or `distinctAvailable` (quantity badge only displays the number of different and available products added to the cart). | `distinct` |
```
Comment (guerreirobeatriz) [inline docs/README.md]:
```suggestion
| `itemCountMode` | `enum` | Quantity badge behavior when displaying the number of total items added in Minicart. Possible values are: `total`  (quantity badge displays the number of items added to the cart), `distinct` (quantity badge only displays the number of different products added to the cart), `totalAvailable`  (quantity badge displays the number of available items added to the cart), and `distinctAvailable` (quantity badge only displays the number of different and available products added to the cart). | `distinct` |
```

**Review by guerreirobeatriz — CHANGES_REQUESTED** [review]

**Review by klzns — CHANGES_REQUESTED** [review]
Thanks for the PR!

It's missing changelog entry

**Comment by gustavopvasconcellos** [comment]
> Thanks for the PR!
> 
> It's missing changelog entry

@klzns It's done men! Thank you

**Review by guerreirobeatriz — APPROVED** [review]

File: react/ProductList.tsx [hunk 5]
```diff
@@ -111,8 +112,37 @@ const ProductGroup: StorefrontFunctionComponent<Props> = props => {
   )
 }
 
+const countModeHandle = (countMode: TotalItemsType, arr: Item[]) => {
```
Comment (victorhmp) [inline react/ProductList.tsx]: Update this function's name to match the change made in https://github.com/vtex-apps/minicart/pull/249

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: react/ProductList.tsx (hunk 5 above)
Comment (gustavopvasconcellos) [reply react/ProductList.tsx]: @victorhmp its done! ;)

**Review by victorhmp — CHANGES_REQUESTED** [review]
@gustavopvasconcellos Everything looks good! But there are lint errors

**Comment by gustavopvasconcellos** [comment]
> @gustavopvasconcellos Everything looks good! But there are lint errors

@victorhmp I see... Can I open another PR to fix this to you? Because these lint errors it isn't because my feature, it's because the outdated code of this app.

I can finish my task with my client then I can fix this on a new PR, what do you think about it? Because each day past the client charging me more because of it...

**Review by klzns — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]

**Pull request merged**